### ISSUE TITLE
New API

### ISSUE THREAD

**Issue by koaning** [koaning/simsity#34]
I _really_ hope this API design is going to be final. 

The service should be fully save-able. 

```python
service = Service(encoder: sklearn.Transformer, indexer:Indexer, data: Union[List, Generator])

service.index(batch_size=100, progress=True)
service.to_disk(folder)
service.query(thing, n_neighbors=100)
Service.from_disk(folder)
```

The thinking here is that I need something like this in order to run a Prodigy recipe that can do something like: `terms.from-simsity` or `image.from-simsity`.

```bash
# To annotate terms
terms.from-simsity <dataset> <service-folder> --seeds "blue,red,green" --label
```

The thinking here is that the `--seeds` can determine the first queries. If none are passed the `--label` is used. This also means that I might want to build a helper class, that can keep on generating candidates for Prodigy.

**Comment by koaning** [comment]
I can also just stop maintaining this and use something like chromadb ...

**Comment by koaning** [comment]
Better idea. How about we just support `hmnslib`? Then the API would become something like:

This would make the API much much more simple. 

```
index = create_index(
    data: Generator, 
    encoder: Union[TransformerMixin, Callable], 
    path:Path, 
    space="l2"
)
index.query()

index = load_index(path)
```

We can use tinyDB for storage. 

```python
from pathlib import Path
from tinydb import TinyDB
from BetterJSONStorage import BetterJSONStorage

path = Path('storage.db')

# with TinyDB(path, access_mode="r+", storage=BetterJSONStorage) as db:
#     db.insert({'int': 1, 'char': 'a'})
#     db.insert({'int': 1, 'char': 'b'})

with TinyDB(path, storage=BetterJSONStorage) as db:
    for i in db:
        print(i)
    print(db.get(doc_id=0))
```
We can use simple queues to generate a stream for Prodigy. 

```python
from queue import SimpleQueue

def stream_from_index(index, seeds, n=10):
    queue = SimpleQueue()
    for seed in seeds:
        queue.put(seed)
    while not queue.empty():
        seed = queue.get()
        items, dist = index.query(seed, n=10)
        for item in items:
            queue.put(item)
            yield item
```
